### ISSUE TITLE
[v. 2.8.2] Connection stuck on daemon/session/watch-cluster-info : Performing connectivity check with timeout 5m0s

### ISSUE THREAD

**Issue by TrojIhor** [telepresenceio/telepresence#2840]
All were fine on 2.7.2. I have the error with 2.8.2 when try to connect.
`telepresence: error: connector.Connect: failed to connect to root daemon: rpc error: code = DeadlineExceeded desc = context deadline exceeded`
Connection stuck on `daemon/session/watch-cluster-info : Performing connectivity check with timeout 5m0s`

I have full access to k8s cluster.
I tried to remove ambassador ns  and reinstall agent, reboot laptop. But it doesn't help.

daemon.log (addresses were hidden by --------------)
```2022-10-27 13:16:40.2248 debug   Logging at this level "debug"
2022-10-27 13:16:40.2279 info    ---
2022-10-27 13:16:40.2280 info    Telepresence daemon v2.8.2 (api v3) starting...
2022-10-27 13:16:40.2280 info    PID is 3872
2022-10-27 13:16:40.2280 info
2022-10-27 13:16:40.2283 debug   Listener opened
2022-10-27 13:16:40.2484 info    daemon/server-grpc : gRPC server started
2022-10-27 13:16:46.0235 debug   daemon/server-grpc/conn=4 : Received gRPC Connect
2022-10-27 13:16:46.0236 info    daemon/session : -- Starting new session
2022-10-27 13:16:46.3265 info    daemon/session : Connected to Manager 2.8.2
2022-10-27 13:16:46.3467 info    daemon/session : also-proxy subnets []
2022-10-27 13:16:46.3467 info    daemon/session : never-proxy subnets [--------------]
2022-10-27 13:16:46.5900 debug   daemon/session/watch-cluster-info : Performing connectivity check with timeout 5m0s
2022-10-27 13:21:46.3578 info    daemon/session : -- Session ended
2022-10-27 13:21:46.3584 error   daemon/session : the port-forward connection to the traffic manager timed out.  The current timeout 5m0s can be configured as "timeouts.trafficManagerConnect" in "/Users/-------/Library/Application Support/telepresence/config.yml"
2022-10-27 13:21:46.5992 debug   daemon/session/watch-cluster-info : Will proxy pods (context deadline exceeded)
2022-10-27 13:21:46.6012 info    daemon/session : Starting Endpoint
2022-10-27 13:21:46.6017 debug   daemon/session/watch-cluster-info : WatchClusterInfo update
2022-10-27 13:21:46.6018 info    daemon/session/watch-cluster-info : Setting cluster DNS to --------------
2022-10-27 13:21:46.6018 info    daemon/session/watch-cluster-info : Setting cluster domain to "cluster.local."
2022-10-27 13:21:46.6019 info    daemon/session/watch-cluster-info : also-proxy subnets []
2022-10-27 13:21:46.6019 info    daemon/session/watch-cluster-info : never-proxy subnets [1--------------]
2022-10-27 13:21:46.6019 info    daemon/session/watch-cluster-info : Adding service subnet --------------
2022-10-27 13:21:46.6020 info    daemon/session/watch-cluster-info : Adding pod subnet --------------
2022-10-27 13:21:46.6020 info    daemon/session/watch-cluster-info : Adding pod subnet --------------
2022-10-27 13:21:46.6020 info    daemon/session/watch-cluster-info : Adding pod subnet --------------
2022-10-27 13:21:46.6020 info    daemon/session/watch-cluster-info : Adding pod subnet --------------
2022-10-27 13:21:46.6020 info    daemon/session/watch-cluster-info : Adding pod subnet --------------
2022-10-27 13:21:46.6021 info    daemon/session/watch-cluster-info : Adding pod subnet --------------
2022-10-27 13:21:46.6021 info    daemon/session/watch-cluster-info : Adding pod subnet --------------
2022-10-27 13:21:46.6021 info    daemon/session/watch-cluster-info : Adding pod subnet --------------
2022-10-27 13:21:46.6022 info    daemon/session/watch-cluster-info : Adding pod subnet --------------
2022-10-27 13:21:46.6022 info    daemon/session/watch-cluster-info : Adding pod subnet --------------
2022-10-27 13:21:46.6022 info    daemon/session/watch-cluster-info : Adding pod subnet --------------
2022-10-27 13:21:46.6022 info    daemon/session/watch-cluster-info : Adding pod subnet --------------
2022-10-27 13:21:46.6022 info    daemon/session/watch-cluster-info : Adding pod subnet --------------
2022-10-27 13:21:46.6022 info    daemon/session/watch-cluster-info : Adding pod subnet --------------
2022-10-27 13:21:46.6023 info    daemon/session/watch-cluster-info : Adding pod subnet --------------
2022-10-27 13:21:46.6023 info    daemon/session/watch-cluster-info : Adding pod subnet --------------
2022-10-27 13:21:46.6023 info    daemon/session/watch-cluster-info : Adding pod subnet --------------
2022-10-27 13:21:46.6023 info    daemon/session/watch-cluster-info : Adding pod subnet --------------
2022-10-27 13:21:46.6023 info    daemon/session/watch-cluster-info : Adding pod subnet --------------
2022-10-27 13:21:46.6024 info    daemon/session/watch-cluster-info : Adding pod subnet --------------
2022-10-27 13:21:46.6024 info    daemon/session/watch-cluster-info : Adding pod subnet --------------
2022-10-27 13:21:46.6024 info    daemon/session/watch-cluster-info : Adding pod subnet --------------
2022-10-27 13:21:46.6024 info    daemon/session/watch-cluster-info : Adding pod subnet --------------
2022-10-27 13:21:46.6024 info    daemon/session/watch-cluster-info : Adding pod subnet --------------
2022-10-27 13:21:46.6024 info    daemon/session/watch-cluster-info : Adding pod subnet --------------
2022-10-27 13:21:46.6024 info    daemon/session/watch-cluster-info : Adding pod subnet --------------
2022-10-27 13:21:46.6025 info    daemon/session/watch-cluster-info : Adding pod subnet --------------
2022-10-27 13:21:46.6064 error   daemon/session/watch-cluster-info : goroutine "/daemon/session/watch-cluster-info" exited with error: PANIC: close of closed channel
github.com/datawire/dlib/dgroup.(*Group).goWorkerCtx.func1.1
	github.com/datawire/dlib@v1.3.1-0.20220715022530-b09ab2e017e1/dgroup/group.go:369
runtime.gopanic
	runtime/panic.go:884
runtime.closechan
	runtime/chan.go:365
github.com/telepresenceio/telepresence/v2/pkg/client/rootd.(*session).onFirstClusterInfo
	github.com/telepresenceio/telepresence/v2/pkg/client/rootd/session.go:498
github.com/telepresenceio/telepresence/v2/pkg/client/rootd.(*session).watchClusterInfo
	github.com/telepresenceio/telepresence/v2/pkg/client/rootd/session.go:470
github.com/telepresenceio/telepresence/v2/pkg/client/rootd.(*session).run.func2
	github.com/telepresenceio/telepresence/v2/pkg/client/rootd/session.go:616
github.com/datawire/dlib/dgroup.(*Group).goWorkerCtx.func1
	github.com/datawire/dlib@v1.3.1-0.20220715022530-b09ab2e017e1/dgroup/group.go:385
github.com/datawire/dlib/derrgroup.(*Group).Go.func2
	github.com/datawire/dlib@v1.3.1-0.20220715022530-b09ab2e017e1/derrgroup/errgroup.go:132
runtime.goexit
	runtime/asm_amd64.s:1594
2022-10-27 13:21:46.6065 info    daemon/session:shutdown_logger : shutting down (gracefully)...
2022-10-27 13:21:55.4044 debug   daemon/server-grpc/conn=8 : Received gRPC Connect
2022-10-27 13:21:55.4045 info    daemon/session : -- Starting new session
2022-10-27 13:21:55.6410 info    daemon/session : Connected to Manager 2.8.2
2022-10-27 13:21:55.6557 info    daemon/session : also-proxy subnets []
2022-10-27 13:21:55.6558 info    daemon/session : never-proxy subnets [--------------]
2022-10-27 13:21:55.8916 debug   daemon/session/watch-cluster-info : Performing connectivity check with timeout 5m0s```

connector.log
```
2022-10-27 13:16:40.4377 debug   Logging at this level "debug"
2022-10-27 13:16:40.4380 debug   Listener opened
2022-10-27 13:16:40.4380 info    ---
2022-10-27 13:16:40.4381 info    Telepresence Connector v2.8.2 (api v3) starting...
2022-10-27 13:16:40.4381 info    PID is 3875
2022-10-27 13:16:40.4381 info
2022-10-27 13:16:40.4560 info    connector/server-grpc : gRPC server started
2022-10-27 13:16:40.6551 debug   connector/server-grpc/conn=1/Connect-1 : called
2022-10-27 13:16:40.6552 info    connector/session : -- Starting new session
2022-10-27 13:16:40.6553 info    connector/session : Connecting to k8s cluster...
2022-10-27 13:16:41.7161 info    connector/session : Server version v1.18.1
2022-10-27 13:16:41.7162 info    connector/session : Context: -------------
2022-10-27 13:16:41.7163 info    connector/session : Server: -------------
2022-10-27 13:16:41.7167 debug   connector/session : Watcher for namespaces in  started
2022-10-27 13:16:41.7172 info    connector/session : Connected to context ------------- (-------------)
2022-10-27 13:16:42.3348 info    connector/session : Connecting to traffic manager...
2022-10-27 13:16:42.3350 error   connector/session : unable to get APIKey: GetAPIKey: not logged in
2022-10-27 13:16:42.3350 debug   connector/session : checking that traffic-manager exists
2022-10-27 13:16:42.3350 debug   connector/session : getHelmConfig
2022-10-27 13:16:42.3352 debug   connector/session : getHelmRelease
2022-10-27 13:16:43.2581 info    connector/session : IsTrafficManager(namespace="ambassador"): current install: version="2.8.2", owner="unknown", state.status="deployed", state.desc="Install complete"
2022-10-27 13:16:43.2582 debug   connector/session : creating port-forward
2022-10-27 13:16:43.2585 debug   connector/session : k8sPortForwardDialer.Dial(ctx, "svc/traffic-manager.ambassador:api")
2022-10-27 13:16:43.7679 debug   connector/session : k8sPortForwardDialer.dial(ctx, Pod./traffic-manager---------------htrdz.ambassador, 8081)
2022-10-27 13:16:43.7680 debug   connector/session : k8sPortForwardDialer.spdyDial(ctx, Pod./traffic-manager---------------htrdz.ambassador)
2022-10-27 13:16:46.0192 debug   connector/session : traffic-manager port-forward established, client was already known to the traffic-manager as "-----"
2022-10-27 13:16:46.0216 info    connector/session : Connecting to root daemon...
2022-10-27 13:21:46.3583 debug   connector/server-grpc/conn=1/Connect-1 : returned
2022-10-27 13:21:46.3584 debug   connector/session : Watcher for namespaces in  stopped
2022-10-27 13:21:51.4290 debug   connector/server-grpc/conn=3/ListCommands-2 : called
2022-10-27 13:21:51.4297 debug   connector/server-grpc/conn=3/ListCommands-2 : returned
2022-10-27 13:21:51.4343 debug   connector/server-grpc/conn=4/Connect-3 : called
2022-10-27 13:21:51.4343 info    connector/session : -- Starting new session
2022-10-27 13:21:51.4344 info    connector/session : Connecting to k8s cluster...
2022-10-27 13:21:51.7066 info    connector/session : Server version v1.18.1
2022-10-27 13:21:51.7066 info    connector/session : Context: -------------
2022-10-27 13:21:51.7067 info    connector/session : Server: -------------
2022-10-27 13:21:51.7067 debug   connector/session : Watcher for namespaces in  started
2022-10-27 13:21:51.7068 info    connector/session : Connected to context -------------(-------------)
2022-10-27 13:21:51.9423 info    connector/session : Connecting to traffic manager...
2022-10-27 13:21:51.9424 error   connector/session : unable to get APIKey: GetAPIKey: not logged in
2022-10-27 13:21:51.9425 debug   connector/session : checking that traffic-manager exists
2022-10-27 13:21:51.9425 debug   connector/session : getHelmConfig
2022-10-27 13:21:51.9425 debug   connector/session : getHelmRelease
2022-10-27 13:21:52.6764 info    connector/session : IsTrafficManager(namespace="ambassador"): current install: version="2.8.2", owner="unknown", state.status="deployed", state.desc="Install complete"
2022-10-27 13:21:52.6765 debug   connector/session : creating port-forward
2022-10-27 13:21:52.6768 debug   connector/session : k8sPortForwardDialer.Dial(ctx, "svc/traffic-manager.ambassador:api")
2022-10-27 13:21:53.1500 debug   connector/session : k8sPortForwardDialer.dial(ctx, Pod./traffic-manager---------------htrdz.ambassador, 8081)
2022-10-27 13:21:53.1501 debug   connector/session : k8sPortForwardDialer.spdyDial(ctx, Pod./traffic-manager---------------htrdz.ambassador)
2022-10-27 13:21:55.4014 debug   connector/session : traffic-manager port-forward established, client was already known to the traffic-manager as "-------------"
2022-10-27 13:21:55.4035 info    connector/session : Connecting to root daemon...
2022-10-27 13:26:55.6679 debug   connector/server-grpc/conn=4/Connect-3 : returned
2022-10-27 13:26:55.6680 debug   connector/session : Watcher for namespaces in  stopped
2022-10-27 13:27:00.7380 debug   connector/server-grpc/conn=6/ListCommands-4 : called
2022-10-27 13:27:00.7382 debug   connector/server-grpc/conn=6/ListCommands-4 : returned
2022-10-27 13:27:00.7422 debug   connector/server-grpc/conn=7/Connect-5 : called
```
traffic manger log
```
 kubectl logs traffic-manager-----htrdz
2022-10-27 08:51:14.0802 info    Traffic Manager v2.8.2 [pid:1]
2022-10-27 08:51:14.0810 info    unable to load license: error reading license: open /home/telepresence/license: no such file or directory
2022-10-27 08:51:14.1004 info    Using cluster domain "cluster.local."
2022-10-27 08:51:14.1033 info    Extracting service subnet ------ from create service error message
2022-10-27 08:51:14.1033 info    Using podCIDRStrategy: auto
2022-10-27 08:51:14.1033 info    Using AlsoProxy: []
2022-10-27 08:51:14.1033 info    Using NeverProxy: []
2022-10-27 08:51:14.1033 info    ExcludeSuffixes: [.com .io .net .org .ru]
2022-10-27 08:51:14.1033 info    IncludeSuffixes: []
2022-10-27 08:51:14.2045 info    Deriving subnets from podCIRs of nodes
2022-10-27 08:51:14.3938 info    Using traffic-agent image "docker.io/datawire/ambassador-telepresence-agent:1.13.1"
2022-10-27 08:51:14.3940 info    prometheus : Prometheus metrics server not started
2022-10-27 08:51:14.6754 info    agent-injector : Loading ConfigMaps from []
2022-10-27 08:51:14.6755 info    agent-injector : Mutating webhook service is listening on :8443
2022-10-27 08:51:15.6764 info    agent-configs : Started watcher for Services cluster wide
2022-10-27 08:51:15.6764 info    agent-configs : Started watcher for ConfigMap telepresence-agents cluster wide
2022-10-27 08:51:15.7351 info    agent-configs : Successfully rolled out ---------
2022-10-27 08:51:15.7813 info    agent-configs : Successfully rolled out ---------
2022-10-27 08:51:15.8291 info    agent-configs : Successfully rolled out ---------
```
cat "/Users/${USER}/Library/Application Support/telepresence/config.yml"
```logLevels:
  userDaemon: debug
  rootDaemon: debug
timeouts:
  agentInstall: 5m
  intercept: 15m
  helm: 5m
  apply: 15m
  clusterConnect: 5m
  proxyDial: 5m
  trafficManagerConnect: 5m
  trafficManagerAPI: 5m
  connectivityCheck: 5m```

Macbook pro (intel): macos Monterey 12.6.1

**Comment by thallgren** [comment]
This log entry looks very suspicious:
```
2022-10-27 13:16:46.5900 debug   daemon/session/watch-cluster-info : Performing connectivity check with timeout 5m0s
2022-10-27 13:21:46.3578 info    daemon/session : -- Session ended
```
The connectivity check is there to check if the cluster's subnets are already routed on your workstation and should typically be very short (a fraction of a second), and if you know that you don't have direct connectivity to the cluster through a VPN (or running your cluster locally), it can even be disabled by setting it to zero. Five minutes just means that you wait that long when there's no connectivity, which is the normal case.

**Comment by thallgren** [comment]
The default connectivityCheck timeout is 500 milliseconds, so the 5 minute value must have been manually configured in your `config.yml`.

**Comment by TrojIhor** [comment]
I connected through vpn. test-vpn didn't pass, interesting. But 2.7.2 passed
```
telepresence test-vpn
Telepresence Daemons have already quit
Please disconnect from your VPN now and hit enter once you're disconnected...

Please connect to your VPN now and hit enter once you're connected...


panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x592c5f6]

goroutine 1 [running]:
github.com/telepresenceio/telepresence/v2/pkg/client/cli.(*vpnDiagInfo).run(0xc00099fc88?, 0xc000932280, {0x60e2985?, 0x60e1193?, 0x0?})
	github.com/telepresenceio/telepresence/v2/pkg/client/cli/cmd_diag_vpn.go:162 +0x956
github.com/telepresenceio/telepresence/v2/pkg/client/cli/output.(*output).runE.func1(0xc000932280, {0x85d2bc0, 0x0, 0x0})
	github.com/telepresenceio/telepresence/v2/pkg/client/cli/output/output.go:91 +0x271
github.com/spf13/cobra.(*Command).execute(0xc000932280, {0x85d2bc0, 0x0, 0x0})
	github.com/spf13/cobra@v1.5.0/command.go:872 +0x694
github.com/spf13/cobra.(*Command).ExecuteC(0xc0007ffb80)
	github.com/spf13/cobra@v1.5.0/command.go:990 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.5.0/command.go:918
github.com/spf13/cobra.(*Command).ExecuteContext(...)
	github.com/spf13/cobra@v1.5.0/command.go:911
main.main()
	github.com/telepresenceio/telepresence/v2/cmd/telepresence/main.go:75 +0x6b3
```

**Comment by thallgren** [comment]
The bug in `telepresence test-vpn` was fixed in #2825. It will be available in the next release.

**Comment by TrojIhor** [comment]
With connectivity check 500ms or 1m connection was successful but cannot intercept with different errors.
500ms: `telepresence: error: rpc error: code = Canceled desc = grpc: the client connection is closing`
1min: `telepresence: error: the server could not find the requested resource (get events.events.k8s.io)`

**Comment by thallgren** [comment]
Set the timeout to 5 seconds (that should be plenty to determine if your workstation already has cluster connectivity), set the daemons loglevels to `debug` in your `config.yml`, then try to connect. Do a `telepresence gather-logs` and provide the logs here so that we can analyze what happens.

**Comment by TrojIhor** [comment]
[telepresence_logs.zip](https://github.com/telepresenceio/telepresence/files/9887271/telepresence_logs.zip)
Got error `telepresence: error: the server could not find the requested resource (get events.events.k8s.io)` with intercept

**Comment by cindymullins-dw** [comment]
HI @TrojIhor , one thing we noticed is it appears you’re on Kubernetes 1.18?  That might be too old for the newest version of Telepresence, [reference here](https://datawire.slack.com/archives/C0418554PRD/p1662734841298949?thread_ts=1662708859.423159&cid=C0418554PRD).  When you upgrade your K8s version could you retest?  We'd also like to check the cluster role RBAC for the traffic manager.  You can do that by running  `kubectl get clusterrole traffic-manager-ambassador -o yaml`.  If you’re one 1.18 you might not see events.k8s.io as below.

- apiGroups:
  - events.k8s.io
  resources:
  - events
  verbs:
  - get
  - watch

**Comment by thallgren** [comment]
Closing this as fixed in 2.9.x.
